stemcell_builder/stages/bosh_ntpd/assets/ntp.conf

```diff
@@ -0,0 +1,13 @@
+driftfile /var/lib/ntp/ntp.drift
+
+statistics loopstats peerstats clockstats
+filegen loopstats file loopstats type day enable
+filegen peerstats file peerstats type day enable
+filegen clockstats file clockstats type day enable
+
+restrict -4 default kod notrap nomodify nopeer noquery
+restrict -6 default kod notrap nomodify nopeer noquery
+
+restrict 127.0.0.1
```

Comment:
On Mon, Nov 30, 2015 at 9:57 AM, Dmitriy Kalinin <notifications@github.com>
wrote:

> In stemcell_builder/stages/bosh_ntpd/assets/ntp.conf
> <https://github.com/cloudfoundry/bosh/pull/1045#discussion_r46177934>:
>
> > @@ -0,0 +1,13 @@
> > +driftfile /var/lib/ntp/ntp.drift
> > +
> > +statistics loopstats peerstats clockstats
> > +filegen loopstats file loopstats type day enable
> > +filegen peerstats file peerstats type day enable
> > +filegen clockstats file clockstats type day enable
> > +
> > +restrict -4 default kod notrap nomodify nopeer noquery
> > +restrict -6 default kod notrap nomodify nopeer noquery
> > +
> > +restrict 127.0.0.1
>
> i thought you guys mentioned that restrict 127... causes ntpd to fail
> actually fetching time from remote servers?
>
I think `restrict 127...` should not cause failures when fetching time from
remote servers.

I have two servers in the NTP pool responding to ~2k NTP queries/sec, and
my VM's clock is able to fetch time from its upstream servers w/out
drifting more than ~20ms.  Here's the config I use:

      # Our upstream timekeepers; thank you Virginia Tech
      server ntp-1.vt.edu
      server ntp-2.vt.edu
      server ntp-3.vt.edu
      server time-b.nist.gov
      # "Batten down the hatches!"
      # see http://support.ntp.org/bin/view/Support/AccessRestrictions
      restrict default limited kod nomodify notrap nopeer
      # Amazon AWS doesn't have IPv6, but I restrict it anyway
      restrict -6 default limited kod nomodify notrap nopeer
      restrict 127.0.0.0 mask 255.0.0.0
      restrict -6 ::1
      # avoid 'out of memory' crash
      rlimit memlock 128
      # require at least 1s between packets from an address, rather than
the default of 2s
      # http://lists.ntp.org/pipermail/questions/2010-April/026306.html
      discard minimum 0

> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cloudfoundry/bosh/pull/1045/files#r46177934>.
>